chips/nrf51/src/radio.rs

```diff
@@ -0,0 +1,456 @@
+pub const PACKET_PAYLOAD_MAXSIZE: u32 = 64;
+pub const PACKET_BASE_ADDRESS_LENGTH: u32 = 4;
+pub const PACKET_STATIC_LENGTH: u32 = 64;
+
+
+// internal Radio State
+pub const RADIO_STATE_DISABLE: u32 = 0;
+pub const RADIO_STATE_RXRU: u32 = 1;
+pub const RADIO_STATE_RXIDLE: u32 = 2;
+pub const RADIO_STATE_RX: u32 = 3;
+pub const RADIO_STATE_RXDISABLE: u32 = 4;
+pub const RADIO_STATE_TXRU: u32 = 9;
+pub const RADIO_STATE_TXIDLE: u32 = 10;
+pub const RADIO_STATE_TX: u32 = 11;
+pub const RADIO_STATE_TXDISABLE: u32 = 12;
+
+
+// static mut TX_BUF: [u8; 16] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16];
+static mut RX_BUF: [u8; 12] = [0x00; 12];
+
+
+
+// FROM LEFT
+// ADVTYPE      ;;      4 bits
+// RFU          ;;      2 bits
+// TxAdd        ;;      1 bit
+// RxAdd        ;;      1 bit
+// Length       ;;       6 bits
+// RFU          ;;      2 bits
+// AdvD         ;;      6 bytes
+// AdvData      ;;      31 bytes
+
+// Header (2 bytes) || Address (6 bytes) || Payload 31 bytes
+static mut PAYLOAD: [u8; 39] = [// ADV_IND, public addr  [HEADER]
+                                0x02,
+                                0xE,
+                                0x00,
+                                // Address          [ADV ADDRESS]
+                                0x90,
+                                0xD8,
+                                0x7A,
+                                0xBD,
+                                0xA3,
+                                0xED,
+                                // [LEN, AD-TYPE, LEN-1 bytes of data ...]
+                                // 0x09 - Local name
+                                // 0x54 0x6f 0x63 0x6b 0x4f 0x54 - TockOs
+                                0x7,
+                                0x09,
+                                0x54,
+                                0x6f,
+                                0x63,
+                                0x6b,
+                                0x4f,
+                                0x53,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00,
+                                0x00]; //[DATA]
+
+pub struct Radio {
+    regs: *const RADIO_REGS,
+    txpower: Cell<usize>,
+    client: Cell<Option<&'static Client>>,
```

Comment:
On second pass, this client actually seems to be entirely unused. Why is it here at all?